app/controllers/application_controller.rb

```diff
@@ -36,5 +50,10 @@ def authenticate_admin!
 
   def authorize
     redirect_to root_path, alert: 'Access denied' unless can? :manage, @question
+  end 
+
+  def reset_session
+      @_request.reset_session
   end
+
```

Comment:
reset_session is a method provided by Active Record, you don't need to define it.  The previous call to reset_session in user_signed_in? will suffice.